config/ols-rbac.yaml

```diff
@@ -0,0 +1,63 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ols-proxy
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: ols-proxy
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: ols-proxy
+subjects:
+- kind: ServiceAccount
+  name: ols-proxy
+  namespace: ols
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ols-proxy
+rules:
+- apiGroups: ["authentication.k8s.io"]
+  resources:
+  - tokenreviews
+  verbs: ["create"]
+- apiGroups: ["authorization.k8s.io"]
+  resources:
+  - subjectaccessreviews
+  verbs: ["create"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ols-proxy-backend-register
+rules:
+- nonResourceURLs:
+    - "/docs"
+    - "/ols"
+    - "/base_llm_completion"
+    - "/feedback"
+  verbs:
+    - get
+    - create
+    - delete
+    - list
+    - update
+    - patch
```

Comment:
how do these verbs relate to the URLs?  What does it mean for a user to have "create" vs "get" permission on "/ols"?

Reply:
Actually those verbs don't add any value when we have `nonResourceURLs` defined. Its just path based action. Since the verb attribute is mandatory i am only keep `get` over there.

Reply:
> Actually those verbs don't add any value when we have nonResourceURLs defined. Its just path based access.

that was my guess

> Since the verb attribute is mandatory I am only keeping get, even a dummy works TBH.

weird that it's required, but given that it is i'd say it's best to use a value that's clearly a noop, rather than a standard verb like `get`.  Some suggestesion: `na', `noop`, `none`

also maybe see what other people have used in this situation.

Reply:
Sorry, I just slipped on my previous comment. Looks like those verbs apply. To cross check this behavior I quickly cloned this [example](https://github.com/brancz/kube-rbac-proxy/tree/master/examples/non-resource-url-token-request) and tried to run it with `noop` verb, it failed with `403 Forbidden`. Below is the mapping for those verbs.

```
get: HTTP GET
create: HTTP POST
delete: HTTP DELETE
list: HTTP GET
update: HTTP PUT
patch: HTTP PATCH
```
I wasn't able to catch this earlier because I was looking at `/docs` and `/healthz` endpoints. They seem to allow all kinds of requests. So to check it properly, I took `/ols` endpoint, enabled `verb:get` and did a POST request to it and ran into `403 Forbidden Error` and the request was not even forwarded to server which is quite expected. 
```
INFO:     Started server process [1]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
```
But when I added `verb:create` I was able to make requests and get legit responses. So keeping only `get` and `create` verbs as they are sufficient for all the endpoints that we have as of now.

Reply:
cool, so basically get==you can do http get on this url, and create==you can do a post on this url.

Reply:
Yes correct.